### ISSUE TITLE
Instance Segmentation training loss not decreasing with with version > 1.7.0

### ISSUE THREAD

**Issue by Abdul-Mukit** [roboflow/rf-detr#1154]
### Search before asking

- [x] I have searched the RF-DETR issues and found no similar bug report.

### Bug

# Problem:
With the rfdetr-1.8.1, instance segmentation training loss isn't decreasing smoothly. In some cases, loss is increasing. No matter which dataset version I use, or pre-trained weight (from my collection or from Roboflow) I start from, the training loss is struggling to go down with 1.8.1. However, this doesn't happen with 1.6.3. 
If I use pre-trained weights from Roboflow, the training loss goes down for several epochs until it starts to struggle. This never happened with 1.6.3. With release-1.8.1, the model is struggling to converge on even my old stable datasets.  

To eliminate the possibility that the root cause might be my dataset or pre-trained weights, I did the following experiments.

Two datasets:
- dataset 0.0.8 (stable)
- dataset 0.0.9 (new dataset)

Two pretrained weights:
- weights20260403 (stable)
- weights20260406 (stable)

RFDETR Version:
- 1.6.3 (stable)
- 1.8.1 (possible issues)

The experiments are named with the pattern:
`<training-date>_dataset<version>_<size><rfdetr-version>_weights<version>`

Anytime rfdetr 1.8.1 is used, the training loss does not decrease properly. Only focusing on the green and blue plots reveals that. The only change between green and blue plots is the rfdetr version number.  

<img width="1852" height="496" alt="Image" src="https://github.com/user-attachments/assets/8c0fff82-b578-4410-9852-adf52c0178d0" />  

<img width="1501" height="473" alt="Image" src="https://github.com/user-attachments/assets/2e4c3e9d-3426-45ba-a054-98853a1f970b" />  

<img width="1853" height="437" alt="Image" src="https://github.com/user-attachments/assets/a7c60b84-6314-4794-99fc-e64265a62f6e" />  

<img width="743" height="420" alt="Image" src="https://github.com/user-attachments/assets/c0fccb1f-ac53-4a62-a2be-1f2c5ef2270c" />   

### Environment

- rfder 1.8.1
- Google Colab
- A100 GPU
- Python 3.12.13
- torch 2.11.0+cu128
- pytorch-lightning 2.6.5

### Minimal Reproducible Example

```
from rfdetr import RFDETRSegMedium

aug_config = {
    "HorizontalFlip": {"p": 0.5},
    "VerticalFlip": {"p": 0.5},
    "Rotate": {"limit": 20, "p": 0.5},
    "Affine": {
        "scale": (0.8, 1.2),
        "translate_percent": (-0.1, 0.1),
        "p": 0.5,
    },
    "ColorJitter": {
        "brightness": 0.5,
        "contrast": 0.5,
        "saturation": 0.5,
        "hue": 0.5,
        "p": 0.5,
    },
}


if pretrain_weights:
    model = RFDETRSegMedium(pretrain_weights=pretrain_weights)
else:
    model = RFDETRSegMedium()


model.train(
    dataset_dir=dataset_dir,
    epochs=10,
    batch_size=16,
    grad_accum_steps=1,
    lr=1e-4,
    num_workers=2,
    device='cuda',
    output_dir=output_dir,
    segmentation_head=True,
    run_test=False,
    tensorboard=True,
    aug_config=aug_config,
    num_queries=100,
    num_select=100
)

```

### Additional

# Concerns:  
* We have grouped disjointed masks. In such cases, COCO-json contains multiple polygons under a single  "segmentation" key. [example comment in supervision](https://github.com/roboflow/supervision/issues/373#issuecomment-1831781340).
* We have masks with holes. [example](https://github.com/cvat-ai/cvat/issues/7030#issuecomment-3646989105)
* Maybe [release-1.7.0](https://github.com/roboflow/rf-detr/releases/tag/1.7.0) had some major changes that are impacting our training.

### Are you willing to submit a PR?

- [x] Yes, I'd like to help by submitting a PR!

**Comment by Borda** [comment]
Hello, with #1159 trying it on this dataset, and it seems to work for me:

<img width="1189" height="571" alt="Image" src="https://github.com/user-attachments/assets/0c875ad6-8f25-437e-b38d-47d28f3a0e36" />

with inference samples:

<img width="1474" height="994" alt="Image" src="https://github.com/user-attachments/assets/096adcca-ce40-448f-a315-1053ac3684fe" />

**Comment by Abdul-Mukit** [comment]
Thank you for looking into this.  
We are definitely observing a performance difference between 1.6.3 and 1.8.1. We used the same dataset. We used same pre-trained weights from Roboflow. The only variation is the release version. With 1.8.1, the loss is remains high.  


<img width="1854" height="484" alt="Image" src="https://github.com/user-attachments/assets/53eb918c-687a-436e-bff6-2434c526fd10" />  

<img width="1847" height="427" alt="Image" src="https://github.com/user-attachments/assets/c5f695c5-3558-4397-a07e-1f938c6d7b83" />  

I'll trigger another training with 1.6.5 and 1.7.0 to see if I can reproduce this issue again. My suspicion is that 1.7.0 may have added a bug. If I find any deeper issues, I'll report back. Thank you.

**Comment by Borda** [comment]
We had to change/fix a bug with loss normalization, which could be the problem...

**Comment by isaacrob** [comment]
@Borda the change with grad accum shouldn't impact a run with GA 1

**Comment by Borda** [comment]
> @Borda the change with grad accum shouldn't impact a run with GA 1

I run a few universe datasets and couldn't reproduce it...

**Comment by Abdul-Mukit** [comment]
I am running with 1.6.5 vs 1.7.0. Will report back tomorrow. Thanks.

**Comment by Abdul-Mukit** [comment]
Version 1.7.0 added the regression.  
Used the same dataset, Roboflow pre-trained weights, training script, and configuration. Then tested 3 versions of rf-detr: `1.6.3`, `1.6.5` and `1.7.0`. Starting from `1.7.0` the training loss is not decreasing properly. Also note the near identical training loss pattern for `1.6.3` and `1.6.5`. This suggests the model can't converge on the training dataset starting from 1.7.0.  

<img width="1838" height="476" alt="Image" src="https://github.com/user-attachments/assets/f3d0f0a2-395c-439d-8d39-f34553b4d8f4" />  

<img width="1848" height="659" alt="Image" src="https://github.com/user-attachments/assets/bd61b65f-5adc-4779-8e6b-9d6686a3acba" />  

Next steps:
- Train/validate on the same small version of our dataset(split).
- Evaluate on the same dataset(split) and find samples with the least mAP.
- Make a dummy dataset with commonplace objects that mimic the annotation scenarios we have in those low-mAP samples.
- Check if I can reproduce the issue so that I can share the dummy dataset here.  

If anyone has any suggestions, that would be appreciated.  

Some unique cases our dataset has, which I suspect might be related:  
- A large item is occluded by a small item on top. 
For example, see option 2 in the issue https://github.com/roboflow/rf-detr/issues/490#issue-3711911433. I use CVAT, which makes that valley almost unnoticeable now (https://github.com/cvat-ai/cvat/issues/7030#issuecomment-3646989105).  
- Annotations with disjointed polygons representing the same Segmentation object. For example, a table has a column of boxes on top, and we have a BEV view. The column of boxes will split the table into two disjointed polygons.

**Comment by Abdul-Mukit** [comment]
<img width="1600" height="486" alt="Image" src="https://github.com/user-attachments/assets/ae6a3210-f72c-43ce-8174-3b1703481072" />  
Resolved, thank you.

**Comment by Borda** [comment]
> Resolved, thank you.

I thank you for your finding 🦝
